pkg/cloudprovider/provider/azure/create_delete_resources.go

```diff
@@ -250,6 +250,58 @@ func getSubnet(ctx context.Context, c *config) (network.Subnet, error) {
 	return subnetsClient.Get(ctx, c.VNetResourceGroup, c.VNetName, c.SubnetName, "")
 }
 
+func getSKU(ctx context.Context, c *config) (compute.ResourceSku, error) {
+	cacheLock.Lock()
+	defer cacheLock.Unlock()
+
+	cacheKey := fmt.Sprintf("%s-%s", c.Location, c.VMSize)
```

Comment:
I actually based this on what we do in AWS to cache AMIs (for example here: https://github.com/kubermatic/machine-controller/blob/master/pkg/cloudprovider/provider/aws/provider.go#L233).

Maybe we do need to think about a more global approach, but in this case I believe caching this information makes sense, because Azure does not let you query individual SKUs, it always makes you query the full list of SKUs to get information for one of them. Thus, those queries are somewhat "expensive" in terms of data transfer and processing them, so I figured caching data in this specific case is better than the alternative. WDYT?

Reply:
I think caching this information at the scope of an individual `Machine` or `MachineDeployment` doesn't make too much sense since the cache would mostly help when you spin up several `MachineDeployments` with the same SKU in short sequence. Once a spec is validated, the `Machine` doesn't need the cached information anymore. And collecting (and expiring?) the data across all `Machines` seems a bit messy to me. It also doesn't need to survive machine-controller restarts :-)